src/buffer.rs

```diff
@@ -745,8 +744,9 @@ where
 
 impl<P, Container> ImageBuffer<P, Container>
 where
-    P: Pixel<Subpixel = u8> + 'static,
-    Container: Deref<Target = [u8]>,
+    P: Pixel + 'static,
+    P::Subpixel: AsBytes,
```

Comment:
I'm very sympathetic towards `zerocopy` but maybe we should hold off from making it part of the public interface for a short while at least. [There is some motion in internals](https://internals.rust-lang.org/t/pre-rfc-safe-transmute/11347) and if the current interface is indeed very stable, it is missing any regards towards `SIMD` and thus the crate could still be in need for a breaking change. Since we faired with only `u8` for a long while, it might be enough to simply define a very minimal (sealed) trait providing impls only for selected primitive types. These `impls` could use `zerocopy` for a safe internals implementation regardless.

Reply:
I left the wrapper trait, `traits::AsBytes`, unsealed because `Primitive` was already unsealed, so one could presumably already use `ImageBuffer` for other types, such as on newtypes over primitives.

Reply:
This leaves `zerocopy` in the public interface, as it is appears as a bound for the provided impls of a public trait. 

I don't quite follow the reasoning with regards to `Primitive`. The two impl blocks guarded by `[P::Subpixel: AsBytes]` only implement `save` and `save_with_format`. Making their requirements strict does not decrease the usability of `ImageBuffer` as a whole but makes it clear which image types *can* be supported for image encoders. A custom pixel type is very unlikely to be encoded correctly and if it truly is byte-level compatible with one of the standard types than you can always create an `ImageBuffer` on the borrowed slice instead.

Reply:
This makes sense. The trait is now sealed as `EncodableLayout`. I'll squash and change the commit message once the rest of the issues are sorted out (if we stick with zerocopy).